Variadic tuples

fredericvauchelles opened the pull request.

fredericvauchelles committed:
Create base rfc

fredericvauchelles committed:
Rename 0000-variadic_tuples.md to 0000-variadic-tuples.md

fredericvauchelles committed:
Update 0000-variadic-tuples.md

fredericvauchelles committed:
Update 0000-variadic-tuples.md

fredericvauchelles committed:
Update 0000-variadic-tuples.md

fredericvauchelles committed:
Update 0000-variadic-tuples.md

fredericvauchelles committed:
Update 0000-variadic-tuples.md

fredericvauchelles committed:
Updated errors

fredericvauchelles committed:
cleaned rfc

pythoneer commented:
[rendered](https://github.com/fredpointzero/rfcs/blob/variadic_tuples/text/0000-variadic-tuples.md)

fredericvauchelles committed:
Uppdated with feedback

fredericvauchelles committed:
Updated from feedback

- Use same syntax for declaration form and expansion form
- Constraint variadic tuple with same arity in declaration

fredericvauchelles committed:
Added note for nested variadic tuples

fredericvauchelles committed:
Updated type-level pattern matching

fredericvauchelles committed:
added todo

fredericvauchelles committed:
Updated RFC from feedbacks

Separated variadic tuple type and variadic tuple
Added destructuration of variadic tuple

fredericvauchelles committed:
Updated syntax of examples

fredericvauchelles committed:
Fixed some syntax issues

fredericvauchelles committed:
Fixed typos

fredericvauchelles committed:
Updated the termination for functions and specialization for recursive impl

fredericvauchelles committed:
Updated errors

fredericvauchelles committed:
Removed fn support, added future possibilities

fredericvauchelles committed:
Added display bounds where required

fredericvauchelles committed:
Always define trait for all tuple and then specialize

fredericvauchelles committed:
Removed '#' char from syntax for clarity

fredericvauchelles committed:
Updated variadic tuple iteration

fredericvauchelles committed:
Fixed missing brackets

fredericvauchelles committed:
Updated syntax to allow tuple merge

fredericvauchelles committed:
Updated RFC with feedbacks

fredericvauchelles committed:
Added support for functions

fredericvauchelles committed:
Updated with trait utilities library

fredericvauchelles committed:
Added future possibility for syntaxic sugar

fredericvauchelles committed:
Fixed syntax issues

fredericvauchelles committed:
Added formal syntax definitions

fredericvauchelles committed:
Fix syntax

fredericvauchelles committed:
Added syntaxic sugar for optional parenthesis

fredericvauchelles committed:
Fixed the confusion about variadic tuple identifiers

fredericvauchelles committed:
Added bound lists in generic parameter groups (future)

fredericvauchelles committed:
Simplified ebnf

fredericvauchelles committed:
Simplified ebnf

fredericvauchelles committed:
Updated with the @ binding pattern

fredericvauchelles committed:
Updated for loop syntax

fredericvauchelles committed:
Updated with optional parenthesis for for loops

fredericvauchelles committed:
Various typo fixes

fredericvauchelles committed:
Fixed feedbacks from comments

fredericvauchelles committed:
Added formatting

fredericvauchelles committed:
Updated generalized type ascription link

fredericvauchelles committed:
Removed dynamic libraries issue

Dynamic libraries are not an issue as machine code for required implementation is embedded in dylibs

fredericvauchelles committed:
Added todo in drawbacks

fredericvauchelles committed:
Updated Bounding variadic tuples in patterns

fredericvauchelles committed:
Updated description

fredericvauchelles committed:
Updated motivation section
